src/Orleans.Transactions/DistributedTM/IClock.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+
+namespace Orleans.Transactions
+{
+    /// <summary>
+    /// System clock abstraction
+    /// </summary>
+    public interface IClock
+    {
+        /// <summary>
+        /// Current time in utc
+        /// </summary>
+        /// <returns></returns>
+        DateTime UtcNow();
```

Comment:
To keep this closer to DateTime.UtcNow you may consider declaring it as a property `DateTime UtcNow { get; }`. Obviously, this is not important. Also perhaps that would be bad style (a getter with side effects).

Reply:
Will leave as is, for the reasons you stated.  We get very ittle from making this a property.